source/globalCommands.py

```diff
@@ -1956,6 +1961,71 @@ def script_reportOrShowFormattingAtCaret(self, gesture):
 		elif repeats == 1:
 			self.script_showFormattingAtCaret(gesture)
 
+	@script(
+		description=_(
+			# Translators: the description for the reportDetailsSummary script.
+			"Report summary of any annotation details at the current location."
+		)
+	)
+	def script_reportDetailsSummary(self, gesture):
+		"""Report the annotation details summary for the single character under the caret or the current nav object.
+		@note: It is tempting to try to report any annotation details that exists in the range formed by prior
+			and current location. This would be a new paradigm in NVDA, and may feel natural when moving by line
+			to be able to more quickly have the 'details' reported. However there may be more than one 'details
+			relation' in that range and we don't yet have a way for the user to select which one to report.
+			For now we minimise this risk by only reporting details at the current location.
+		"""
+		log.info("Report annotation details summary at current location.")
+		# Common cases, vbuf available: Eg editable text, or regular web content
+		try:
+			caret: textInfos.TextInfo = api.getCaretPosition()
+		except ValueError:
+			log.debugWarning("Unable to get the caret position.", exc_info=True)
+			return
+		caret.expand(textInfos.UNIT_CHARACTER)
+		nvdaObject: Optional[NVDAObject] = caret.NVDAObjectAtStart
+		log.debug(f"Trying with nvdaObject : {nvdaObject}")
+		# So far NVDAObjectAtStart works in all situations tested.
+		#
+		# An alternative might be to:
+		# - Iterate over the FieldCommands looking for a ControlField with the "hasDetails" attribute.
+		# Example:
+		# for f in ti.getTextWithFields():
+		# 	if(
+		# 		f
+		# 		and isinstance(f, textInfos.FieldCommand)
+		# 		and isinstance(f.field, textInfos.ControlField)
+		# 		and f.field.get("hasDetails")
+		# 	):
+		#
+		# - Use controlIdentifier_* or UniqueID (keys in the FieldCommand's ControlField) to create an
+		# IAccessible NVDAObject. Note, in focus mode UniqueID will only contain one value (the
+		# controlIdentifier_ID) the docHandle will need to be fetched another way.
+		# Example:
+		#
+		# docHandle = f.field.get("controlIdentifier_docHandle")
+		# ID = f.field.get("controlIdentifier_ID")
+		# objWithDetails = IAccessible.getNVDAObjectFromEvent(docHandle, winUser.OBJID_CLIENT, ID)
+		annotation: Optional[str] = nvdaObject.detailsSummary
+		if annotation:
+			log.debug("NVDAObjectAtStart of caret has details.")
+		elif api.getNavigatorObject():

```

Comment:
In which cases is it necessary to fall back to the navigator objects and why you prefer to use it rather than `api.getFocusObject`?

Reply:
I agree. If caret position is used instead of review position, then focus object should be used instead of navigator object.

Reply:
So that I can add a comment here about why, do either of you have an explanation?

Reply:
I'm not sure what you're asking here. If the question  is: "Why should I use focused object rather than navigator object?" then the answer has been given by @michaelDCurran  above. However if you're asking "Why use a fallback when there are no annotations at the caret position?" then I cannot answer that one and was hoping you can explain in which situations there were no annotations at the caret position yet they were present on the navigator object.

Reply:
Do you mean this comment:
>I agree. If caret position is used instead of review position, then focus object should be used instead of navigator object.

This seems to imply that review position and navigator object are some kind of semantic pair, and caret and focus are another semantic pair. Is that true?

I'd like to include the reasoning (the answer to "why is it done this way?") in the code comments. This helps future developers understand the intent, the use-cases, and helps to prevent regressions.

Unfortunately I don't feel like I have a good understanding of this. The comments so far are telling me what this should do without explaining why that is important. 

If I had to guess, it's because of settings like `Review cursor follows focus` which when disabled may make using the navigator object unreliable. Though I don't know that for sure, how is the navigator object related to this setting that talks about the review cursor?

Reply:
> This seems to imply that review position and navigator object are some kind of semantic pair, and caret and focus are another semantic pair. Is that true?

That is one way to look at it. Essentially when moving the review cursor you're moving the navigator object. This is a slight oversimplification, since if you are in object review the navigator object stays the same but in both screen review and document review navigator object follows the review cursor position. The relation between navigator object and review cursor is however not as important as having an understanding as to why we need a fallback at all. Could you please explain why you've written this section of code in the first place. Ideally the code comment here should explain when the fallback (regardless if it is to the focused or navigator object) is needed.

Reply:
I agree a better comment is required here, this is supposed to be for the corollary of:
> `# Common cases, vbuf available: Eg editable text, or regular web content`

I.E. This section of code was intended for when the virtual buffer is not available, given the other changes I'm not actually sure it is still required. I believe there is an example when using Google Sheets, however right now I can't recreate it. I'll have to talk to @michaelDCurran after the break about the exact case. We should also add a test for this case.